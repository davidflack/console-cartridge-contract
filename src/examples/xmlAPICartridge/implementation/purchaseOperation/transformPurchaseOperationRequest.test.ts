import {
  canOfFreshDogKibbleFixture,
  dogTuxedoFixture,
  fancyCatCollarFixture,
} from './__fixtures__';
import {
  generateProductPurchaseElementFromItem,
  transformPurchaseOperationRequest,
} from './transformPurchaseOperationRequest';

describe(generateProductPurchaseElementFromItem, () => {
  it('generates an xml element to correspond to our purchase item', () => {
    expect(
      generateProductPurchaseElementFromItem({
        quantity: 1,
        item: fancyCatCollarFixture,
      }),
    ).toMatchSnapshot();
  });
});

describe(transformPurchaseOperationRequest, () => {
  it('generates an xml payload for our purchase', () => {
    const id = 'our-transaction-id';
    expect(
      transformPurchaseOperationRequest({
        id,
        items: [
          {
            quantity: 1,
            item: fancyCatCollarFixture,
          },
          {
            quantity: 3,
            item: canOfFreshDogKibbleFixture,
          },
          {
            quantity: 1,
            item: dogTuxedoFixture,
          },
        ],
      }),
    ).toMatchSnapshot();
  });
});
