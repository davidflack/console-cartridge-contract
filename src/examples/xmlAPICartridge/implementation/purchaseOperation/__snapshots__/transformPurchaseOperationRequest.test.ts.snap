// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateProductPurchaseElementFromItem generates an xml element to correspond to our purchase item 1`] = `
"<product>
  <sku>61b04f7f-2682-4206-b2e1-0711537073a6</sku>
  <quantity>1</quantity>
</product>"
`;

exports[`transformPurchaseOperationRequest generates an xml payload for our purchase 1`] = `
"
<?xml version='1.0' encoding='UTF-8'?>
<purchase>
  <id>our-transaction-id</id>
  <products>
    <product>
  <sku>61b04f7f-2682-4206-b2e1-0711537073a6</sku>
  <quantity>1</quantity>
</product>
<product>
  <sku>d188d07a-1e5a-4afc-a481-1eb1450da38f</sku>
  <quantity>3</quantity>
</product>
<product>
  <sku>9816f1be-554d-11ed-bdc3-0242ac120002</sku>
  <quantity>1</quantity>
</product>
  </products>
</purchase>
"
`;
